src/AutoRest.CSharp.V3/Input/Source/SourceInputModel.cs

```diff
@@ -3,31 +3,120 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
 using System.Linq;
+using Azure.Core;
+using Microsoft.CodeAnalysis;
 
 namespace AutoRest.CSharp.V3.Input.Source
 {
     public class SourceInputModel
     {
-        public SourceInputModel(
-            ModelTypeMapping[] definedSchemaTypes,
-            ClientTypeMapping[] definedClientTypes)
+        private readonly Compilation _compilation;
+        private readonly INamedTypeSymbol _schemaNameAttribute;
+        private readonly INamedTypeSymbol _clientAttribute;
+        private readonly INamedTypeSymbol _schemaMemberNameAttribute;
+
+        public SourceInputModel(Compilation compilation)
+        {
+            _compilation = compilation;
+            _schemaNameAttribute = compilation.GetTypeByMetadataName(typeof(CodeGenModelAttribute).FullName!)!;
+            _schemaMemberNameAttribute = compilation.GetTypeByMetadataName(typeof(CodeGenMemberAttribute).FullName!)!;
+            _clientAttribute = compilation.GetTypeByMetadataName(typeof(CodeGenClientAttribute).FullName!)!;
+
+            var assembly = _compilation.Assembly;
```

Comment:
Nit: other instances of improper `var` (to note: I'm a fan of `var` - just towing the line 😉).